src/resources/Resource.ts

```diff
@@ -1,8 +1,18 @@
-import { parse as parseUrl } from 'url';
 import List from '../data/list/List';
 import Maybe from '../types/Maybe';
 import TransformingNetworkClient from '../TransformingNetworkClient';
 
+/**
+ * Returns the parsed search parameters from the passed URL. For example: `'https://example.com?id=5'` is converted to
+ * `{ id: 5 }` (and `'https://example.com'` is converted to `{}`).
+ *
+ * If multiple parameters have the same key (`'https://example.com?id=5&id=6'`), exactly one of them will be
+ * represented in the returned object.
+ */
+function parseQueryInUrl(url: string) {
+  const result: Record<string, string> = {};
+  return new URL(url).searchParams.forEach((value, key) => result[key] = value), result;
```

Comment:
Code style nit. I think splitting this expression over 2 lines will be more commonly understood.

```suggestion
  new URL(url).searchParams.forEach((value, key) => result[key] = value);
  return result;
```

Reply:
Sure!